### DISCUSSION TITLE
Loop Builder Filtering Suggestions and Woocommerce Gallery images

### DISCUSSION THREAD

**Discussion by awebguy99** [elementor/elementor#21203] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

**Loop builder filtering**
1. Currently there are no filtering system for the loop builder. I know you're working on this but I'd like to make suggestions so it's done right.

**Woocommerce loop item dynamic Product Gallery image**
2. Currently when we're building our loop item, we can choose just the Product Image but not a Product Gallery image for the product.

### Describe the solution you'd like

**Loop builder filtering**
Please make the filtering as least as good and have the same number of filtering types as the many Woocommerce filtering plugins out there.
Currently I use JetSmartFilters, which lets me have these filter types;
- Checkbox List (very useful for product categories)
- Select
- Range (very useful for price range, so you can drag the price to only show products from $100 to $250 for example)
- Date range
- Date period
- Radio select
- Rating
- Alphabet (actually, I've never used this!)
- Search
- Visual (great for product Brands)

For your filtering system to be useable (at least to the point where a lot of people can stop using 3rd party filtering plugins), it really needs to have at least most of these options.
Just look at most good e-commerce stores and they will have a filtering system with some or most of these options. 

Other important filtering features:
1) Also, when a filter is selected it needs to dynamically change the URL (I think via AJAX) to included the filtering change. This will let users copy the URL and send to friends, post on social sites etc. When someone uses that URL, it opens the archive with those filters already selected.
E.G. If I find all the t-shirts in my size, between $100 and $250, someone clicking the URL I send them will automatically see the same selection of t-shirts.

2) And regarding this URL, ideally it would use friendly naming.
E.G. Not good would be /domain/products?filter=dh3u3h3?tc=h4u4uh4uh4
Good would be something like /domain/products?t-shirts/color:red/size:34to45

**Woocommerce loop item dynamic Product Gallery image**
It would add more interactivity to the user experience if Product Gallery images were a choice in the Dynamic options in loop items. This would allow various things but one example is we could have the main Product Image, then on the Hover tab, choose a Gallery Image.
Or we could even put the Gallery Images somewhere else in the loop item.

### Describe alternatives you've considered

I have to use 3rd party plugins like JetSmartFilters and JetWoobuilder.

**Comment by awebguy99** [comment]
I don't know if we're allowed to bump but getting a new filtering feature done right is going to be extremely important for Elementor. If it doesn't have the features of the current 3rd party filtering plugins, me and a high % of Elementor web designers simply won't use it and still have to use those 3rd party plugins.

**I'm asking the community to get behind this feature request. Please consider up voting this.**

**Reply by stefckee** [reply]
Yes, they should def!

**Comment by hitsgee** [comment]
JetSmart Filters does not yet work with the new loops etc only the classic post widget. Big headache for us.

**Reply by stefckee** [reply]
Yup. Or the listing grid builder. We really need filters for the loop grid.

**Comment by awebguy99** [comment]
Thanks for the support @stefckee and @hitsgee.

**Any others? Come on everyone, we need our web designer community to get behind this feature request please**, so the Filtering they're developing for the Loop Builder is **done right**. My great concern is, they'll do 50% to 80% of the needed features (I.E. 50% to 80% of JetSmartFilter features), then me and a large % of us won't use it, and be stuck with having to use a 3rd party filtering plugin.

Filtering is such an essential feature to get right on e-commerce stores (it can literally make the store owner more money if done right) and also other sites (such as car/house listing etc).

**Comment by 79ho3ein** [comment]
In my opinion, Elementor should work on a professional filtering system to provide the filters mentioned by @awebguy99. We know now elementor works on loop filter bar. But most likely, this filter will be implemented in the loop grid widget and will only filter the content of the loop based on tags and categories.
I hope Elementor will work on advanced filter system soon.

**Comment by awebguy99** [comment]
Thank you @79ho3ein 
It will make Elementor exponentially better. :)

**Comment by alsage13** [comment]
This is HUGE for me. I have a site right now that needs this feature in about 6 different places, and since I'm not comfortable with writing my own queries, I'm having to pay a developer to do so. What I'm looking for is twofold:

- Filter by a specific ACF field (usually by a certain response from a select box or radio button)
- Filter by an ACF date field, specify upcoming and past posts (similarly to the others, I am working on a site with a lot of events and I'd like to be able to easily show upcoming events, and those that have already passed.)

I'm sure I can come up with additional uses, but these two are enormous for me!

**Comment by hitsgee** [comment]
Crocoblock are releasing an update this week with support for loop grid with jet smart filters. There is already a beta for this on GitHub but it currently does not work with current query so not great for archive use.

**Reply by awebguy99** [reply]
Thanks for the update. While this will hopefully give us the ability to use Jetsmart filters with the new Elementor loop builder, what I (and I'm guessing most other web designers/developers) really want are these filter features available in the **core** Elementor loop builder.

EDIT: I just wanted to say, I have nothing against the Jet developers, they seem like a great bunch of people wanting to make a great product. I just want to use as little 3rd party plugins as possible. These features should be available in core. If you release a loop builder, you really need to release a filtering system, they go hand in hand.

**Comment by toineenzo** [comment]
That's a nice list of suggestions! A list that Elementor should ABSOLUTELY implement if they want their new taxonomy filter to be actually useful

**Comment by threeforces** [comment]
I hereby push the possibility to integrate product galleries into the loops. I would find it practical to be able to insert them just as dynamically as the product images, e.g. as the background of a container. This would make the loops even more dynamic.

Currently product image and category image exist as dynamic attribute. An additional function for the product gallery would be great.

**Reply by awebguy99** [reply]
Thank you. Yes, adding the product gallery as a dynamic content option would open up a load of design options and would even allow me to stop using another 3rd party plugin.

**Comment by nicholaszein** [comment]
Hello @awebguy99,

Thanks for contributing your suggestion! 🙏

Your ideas and input as a Web Creator are valuable to us. We're always keen to explore ways to enhance Elementor to meet our community's needs better.

### 💡 Due to the high volume of feature requests we receive and our commitment to continuous improvement, we can't guarantee that every suggestion will be immediately included in our roadmap.

That being said, your idea is important to us. We'll keep your suggestion on our radar, and if it attracts a significant number of votes in the future, we'll consider implementing it.

We appreciate your patience as we take the time to review your request thoroughly. We aim to ensure all decisions align with our mission to provide Web Creators with the tools they need to build their dream websites.

----

### 🗃️ Please understand that while we do welcome new feature requests, we can't track multiple requests per thread efficiently.

I'll leave this request open because it already has a lot of votes. But in the future, please submit only ☝️ **_<ins>one request per thread</ins>_** so we can track them better and allow users to vote only for the features they want to see implemented.

Kind regards

**Comment by awebguy99** [comment]
Thanks @nicholaszein

Would this work?;
1. I remove the 2nd request ('Woocommerce loop item dynamic Product Gallery image') from this post.
2. Create a new post for the 'Woocommerce loop item dynamic Product Gallery image' request.

**Reply by threeforces** [reply]
created a seperate request: https://github.com/orgs/elementor/discussions/23737
